web/source/kmwuibutton.ts

```diff
@@ -213,15 +214,21 @@ if(!window['keyman']['ui']['name']) {
         } 
         else
         {
-          kbdId.className = osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
+          let osk = keymanweb.osk;
+          kbdId.className = osk && osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
         }
       }
     }  
   
-    /**
-     * UI Functions called by KeymanWeb or OSK
-     */     
-    osk['addEventListener']('show',
+    ui.registerEvents = function() {
+      let osk = keymanweb.osk;
+      if(!osk) {
+        return;
+      }
```

Comment:
Could we have a race if this is called after osk initialization now, and so it never gets its events registered?

Reply:
I don't believe so.  Note that `ui.registerEvents` is called toward the end of UI module initialization, which itself is after OSK initialization.  It's only referenced within the UI module.

https://github.com/keymanapp/keyman/blob/8b8c32cf6d040a1aea0c22d824453d391c40d841/web/source/dom/domManager.ts#L1588-L1598

The last line of which is what calls the UI module init.